### ISSUE TITLE
Ability to set a retention policy for clearing data

### ISSUE THREAD

**Issue by bsclifton** [brave/brave-browser#1059]
Related to https://github.com/brave/browser-laptop/issues/7890

## Description
While https://github.com/brave/brave-browser/issues/492
 captures clearing on exit, users should be able to go further and set a retention policy for history / cookies / etc. 

For example, maybe user can choose to clear after:
- 3 days
- 1 week
- 1 month
- custom

Safari offers an option like this:
![image](https://user-images.githubusercontent.com/4733304/45262702-e452c800-b3d0-11e8-8dd8-e836d866d540.png)

**Comment by ShivanKaul** [comment]
@bsclifton - we talked about this in privacy confab today. Some part of this could be UX for 1P ephemeral storage work (but not the history clearing), which would also give global + per-site toggles. If there's interest, we should come up with a spec.

**Comment by Brave-Matt** [comment]
Wanted to hop in here — users often request the option to hold onto their browsing history for longer than the 90 days that Chromium defaults to. We have a feature request open on Community for this as well. Might be worth taking another look at:
https://community.brave.com/t/extend-the-amount-of-stored-browsing-history/331674
https://community.brave.com/t/detailed-history-view/162931
https://community.brave.com/t/brave-android-automatically-deletes-browsing-history-after-3-months/342423/2

**Comment by GoldenMaximo** [comment]
Bump.

**Comment by dpthegrey** [comment]
yes we need to keep more history record such that suggestions come directly from locally stored URL and Bookmarks when searching instead of hitting search engine. One less step in reaching to the website we want. Would love to see this feature coming.

**Comment by jkdn0921s** [comment]
bump too. 
i am interested as well to keep the browsing history as long as i need it (custom).
This is the reason why I had to move back to Firefox bcz with Firefox I can keep my history and it does get auto-deleted :|

**Comment by evankh** [comment]
Bump. What's really annoying is that there's no warning, notification, or setting anywhere that tells you it's even a possibility. I've lost my game history on a certain daily game site twice now - it's stored in an IndexedDB, and apparently that site data gets wiped out too. And it's not just the games that were older than the 3 month limit: as soon as the oldest game passed that mark, the entire database disappeared.

If it were just the page history being deleted, that would be pretty annoying but I could live with it; wiping all data for a site I visit every day is incredibly frustrating.

**Comment by codetheron** [comment]
Same here. There are extensions that allows to store history longer than 90 days but this setting should be built in into browser.

**Comment by rom0808** [comment]
Bump, it would be very useful. Even Vivaldi add it now. You could add the same type of menu. Thanks.

**Comment by Deagle2** [comment]
Bump

**Comment by jkdn0921s** [comment]
Yeah agree, Vivaldi works just fine on this and retains it. To be honest, bcz of this single reason, I went back to Firefox and I barely use Brave nowadays. I use Firefox daily bcz i also know if i need to change PCs or change OSs, i got all the data backed up there. Not to even mention I can see history way way way back through Firefox synced profile too.

**Comment by qahtni** [comment]
Bump, 90 days is too short.

**Comment by jkdn0921s** [comment]
bump, any update on this pretty please ? 
Thank you!

**Comment by snoopiedoom** [comment]
bipitty bumpetty

**Comment by stefanolomo** [comment]
____                        
|  _ \                       
| |_) |_   _ _ __ ___  _ __  
|  _ <| | | | '_ ` _ \| '_ \ 
| |_) | |_| | | | | | | |_) |
|____/ \__,_|_| |_| |_| .__/ 
                                 | |    
                                 |_|    

Any updates?

**Comment by raldisk** [comment]
bump! any updates?

**Comment by glwi-commits** [comment]
BUMP PLEASE
THIS IS HOW I FIND OUT??? WHEN A HOSTED MEDIA LINK I CARE ABOUT IS GONE????
LET ME STORE MY HISTORY LOCALLY FOR MORE THAN 90 DAYS. 
BUMP BUMP BUMP

**Comment by rickyelopez** [comment]
> BUMP PLEASE THIS IS HOW I FIND OUT??? WHEN A HOSTED MEDIA LINK I CARE ABOUT IS GONE???? LET ME STORE MY HISTORY LOCALLY FOR MORE THAN 90 DAYS. BUMP BUMP BUMP

while I don't agree with the tone I definitely agree with the sentiment.

I've resorted to using the History Trends Unlimited addon so that I definitely won't lose any history, however, typing stuff into the address bar doesn't search the plugin's index. So if something doesn't come up when I type it in the address bar, I then have to open the plugin's interface separately and search there manually. Not great.
